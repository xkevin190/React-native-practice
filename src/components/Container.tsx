import * as React from 'react';
import {StyleSheet, SafeAreaView, View, Platform} from 'react-native';
import { colors } from '../constants/colors';

interface ContainerProps {
  children: JSX.Element;
}

const Container = (props: ContainerProps) => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.container}>
        <>{props.children}</>
      </View>
    </SafeAreaView>
  );
};

export default Container;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
});
