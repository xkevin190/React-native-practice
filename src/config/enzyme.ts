import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
