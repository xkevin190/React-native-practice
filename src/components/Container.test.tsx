import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Container from './Container';
import {View} from 'react-native';

describe('Container (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Container>
        <View testID="children" />
      </Container>,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('children should be defined', () => {
    const result = wrapper
      .find('View')
      .at(1)
      .findWhere(node => node.prop('testID') === 'children')
      .exists();
     expect(result).toBe(true);
  });
});
