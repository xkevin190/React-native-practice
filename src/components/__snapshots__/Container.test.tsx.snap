// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container (Unit-Test) general functionality should match snapshot 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#17141f",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#17141f",
        "flex": 1,
      }
    }
  >
    <View
      testID="children"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`Container (Unit-Test) should match snapshot 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#17141f",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#17141f",
        "flex": 1,
      }
    }
  >
    <View
      testID="children"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
